### PR TITLE
Add a large object scenario for the CLI based tests in object storage…

### PR DESCRIPTION
**Pull request by stfeng**
… service benchmark. (the default is still old behavior)

### CODE REVIEW

**Comment by stfeng** [comment]
@yuyantingzero please take a look. Thanks!

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 1]
```diff
@@ -59,6 +59,11 @@
                   'api_namespace: runs API based benchmarking for namespace '
                   'operations.')
 
+flags.DEFINE_enum('cli_size', 'normal',
+                  ['normal', 'large'],
+                  'size of the cli tests. Normal means half small and half \n'
+                  'large files. large means all large files each with 1GB.')
```
Comment (thobrla) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: "large files" has two fairly different meanings here, yes?

For 'normal', a "large file" is 32MB.  For 'large', a "large file" is 1GiB.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 2]
```diff
@@ -59,6 +59,11 @@
                   'api_namespace: runs API based benchmarking for namespace '
                   'operations.')
 
+flags.DEFINE_enum('cli_size', 'normal',
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: This name is a bit confusing. Maybe cli_test_size.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 3]
```diff
@@ -477,14 +493,18 @@ def _CliBasedTests(output_results, metadata, vm, iteration_count,
                                 'iterations.')
 
   # Report various percentiles.
+  metrics_prefix = ''
+  if FLAGS.cli_size != 'normal':
+    metrics_prefix = '%s ' % FLAGS.cli_size
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Seems like you want to keep the default results format compatible with old ones. I suggest adding a field in metadata cli_test_size: normal/large, which is more consistent. Up to you.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py [hunk 4]
```diff
@@ -107,7 +114,10 @@
 
 UPLOAD_THROUGHPUT_VIA_CLI = 'upload throughput via cli Mbps'
 DOWNLOAD_THROUGHPUT_VIA_CLI = 'download throughput via cli Mbps'
+
 CLI_TEST_ITERATION_COUNT = 100
+LARGE_CLI_TEST_ITERATION_COUNT = 20
+
 CLI_TEST_FAILURE_TOLERANCE = 0.05
 # Azure does not parallelize operations in its CLI tools. We have to
```
Comment (yuyantingzero) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Someone mentioned a while ago Azure supports parallelize operations nowadays. Maybe add that in next PR?

**Comment by yuyantingzero** [comment]
I noticed there is no large cli test for Azure, any reason for that?

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 4 above)
Comment (stfeng) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: sure, that would be a separate PR.

**Comment by stfeng** [comment]
Added support for Azure with large cli scenario. Missed it in previous CL. Thanks for pointing this out @yuyantingzero !

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 3 above)
Comment (stfeng) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: I thought about this, but that means if existing dashboards based on the metrics names would be broken unless we change those dashboards to also look for the new metadata field (cli_test_size). I'd prefer to add new metric names here so those existing dashboards continue to work.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 2 above)
Comment (stfeng) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Done.

File: perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py (hunk 1 above)
Comment (stfeng) [inline perfkitbenchmarker/benchmarks/object_storage_service_benchmark.py]: Good point, clarified the help text here.

**Comment by yuyantingzero** [comment]
LGTM
